vhost/src/vhost_user/connection.rs

```diff
@@ -236,6 +236,31 @@ impl<H: MsgHeader> Endpoint<H> {
         Ok(())
     }
 
+    /// Send a message with header and body. Optional file descriptors may be attached to
+    /// the message.
+    ///
+    /// # Return:
+    /// * - number of bytes sent on success
+    /// * - SocketRetry: temporary error caused by signals or short of resources.
+    /// * - SocketBroken: the underline socket is broken.
+    /// * - SocketError: other socket related errors.
+    /// * - PartialMessage: received a partial message.
+    pub fn send_message_oversized<T: ByteValued>(
```

Comment:
Ditto, new API that are supposed to be used only for GPU protocol, should not be added and exposed with the vhost-user API. It's really confusing for future user why we have send_message and send_message_oversized.

Should be okay if we have a way to check if the device support it or not and return an error in that case.

Reply:
I also suggest adding the API for "oversized" message all together in a separate commit with a proper description in the commit message and also in the code comment.

Reply:
yeah, I agree. Since this methods are specific for the GPU protocol, maybe we can put it behind a gpu feature.

Reply:
I'm not sure a feature is enough. What I mean is that if these functions are not meant to be used by any device, then I would avoid putting in the common modules, but in a gpu-specific module, otherwise device implementers can be confused and use this API.

Otherwise we have to at least put a check of the negotiated virtio features and return an error if this API is not supported.

Reply:
`Endpoint` is internal to this crate, it is not a public API, so this shouldn't really be a problem.

As far as I can tell there is not really a message size limit in the protocol itself, this is just an internal limitation of `Endpoint` implementation, see this comment:
https://github.com/rust-vmm/vhost/blob/bd6bf1325b4930d4123f6154ce755bd5289c617b/vhost/src/vhost_user/message.rs#L26-L33

Ideally I think it could be nice to not have this limit and handle arbitrary sized messages, but I didn't want to rewrite the whole `Endpoint` just for this.  If the other side (a frontend) of the  vhost user gpu socket protocol is implemented in this crate, this would likely be necessary. Since this is an internal API, I hope this would be fine for now.

Reply:
okay, I see.

I don't remember why we have that limit on the sender side. On the receiver side it looks like it's a limit to avoid pre-allocation, but IMHO we can remove that limit in the sender. If the receiver doesn't like the size, can always send back an error.

I mean, we can just check if the len is on 32bit (as you did), without adding new API. WDYT?

Reply:
Yes sure, I can rewrite it like that. I maybe see one reason it is the way it is: for testing the backend against fronted in this crate, this might have been like an internal assertion when writing tests (testing that the test itself is OK) But this isn't really that useful, since the receiving side would hopefully fail anyway.